src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -1645,6 +1645,36 @@ public bool Equals(StringBuilder sb)
             }
         }
 
+        /// <summary>
+        /// Determines if the contents of this builder are equal to the contents of ReadOnlySpan<char>.
+        /// </summary>
+        /// <param name="value">The ReadOnlySpan<char>.</param>
+        public bool Equals(ReadOnlySpan<char> value)
+        {
+            if (value.Length != Length)
+                return false;
+
+            StringBuilder thisChunk = this;
+            int thisChunkIndex = thisChunk.m_ChunkLength;
+
+            for (int i = Length - 1; i >= 0; i--)
+            {
+                --thisChunkIndex;
+                while (thisChunkIndex < 0)
+                {
+                    thisChunk = thisChunk.m_ChunkPrevious;
+                    if (thisChunk == null)
+                        break;
+                    thisChunkIndex = thisChunk.m_ChunkLength + thisChunkIndex;
+                }
+
+                if (thisChunk.m_ChunkChars[thisChunkIndex] != value[i])
```

Comment:
It may be more efficient to lift this field access `thisChunk.m_ChunkChars` into a local variable ie create `thisChunkChars` .. if you benchmark this, you can try it.